src/Database/Redis/Schema/Lock.hs

```diff
@@ -39,24 +41,51 @@ import qualified Data.ByteString.Char8 as BS
 import System.Random    ( randomIO )
 
 import Control.Concurrent  ( threadDelay, myThreadId )
+import Control.Exception   ( Exception )
 import Control.Monad.Fix   ( fix )
 import Control.Monad.Catch ( MonadThrow(..), MonadCatch(..), MonadMask(..), throwM, finally )
 import Control.Monad.IO.Class ( liftIO, MonadIO )
 
 import qualified Database.Redis.Schema as Redis
 
+data ExclusiveLockAcquireTimeout = ExclusiveLockAcquireTimeout
+  { elatRefIdentifier :: Redis.SimpleValueIdentifier
+  , elatLockParams :: LockParams
+  , elatOurId :: LockOwnerId
+  }
+  deriving (Show)
+
+instance Exception ExclusiveLockAcquireTimeout
+
+data ShareableLockAcquireTimeout = ShareableLockAcquireTimeout
+  { slatRefIdentifier :: ByteString
+  , slatLockParams :: ShareableLockParams
+  , slatSharing :: LockSharing
+  , slatOurId :: LockOwnerId
+  }
+  deriving (Show)
+
+instance Exception ShareableLockAcquireTimeout
+
+-- TODO(daan): also include lock value with exceptions?
+
 data LockParams = LockParams
   { lpMeanRetryInterval :: NominalDiffTime
   , lpAcquireTimeout    :: NominalDiffTime
   , lpLockTTL           :: Redis.TTL
   }
+  deriving (Show)
 
 -- | ID of the process that owns the Redis lock.
 newtype LockOwnerId = LockOwnerId { _unLockOwnerId :: ByteString }
+  deriving (Show)
```

Comment:
This could probably be `deriving newtype Show` without introducing ambiguity. Generally, I think it's nice to avoid things like `UserRow { id = UserRowId { unUserRowId = DatabaseId { unDatabaseId = 1234 } } }` in favour of  `UserRow { id = 1234 }`.

Either way, I'd suggest using an explicit deriving strategy here, like above.

Reply:
Personally, I don't prefer that, because I want the `Show` instance to make it clear that we're dealing with a newtype instead of some more primitive type; more verbose output is then annoying, but worth it IMO.

Reply:
Sure!